ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt

```diff
@@ -111,4 +111,10 @@ internal class OkHttpWebsocketSession(
 }
 
 @Suppress("KDocMissingDocumentation")
-class UnsupportedFrameTypeException(frame: Frame) : IllegalArgumentException("Unsupported frame type: $frame")
+class UnsupportedFrameTypeException(
+    private val frame: Frame
+) : IllegalArgumentException("Unsupported frame type: $frame"), CopyableThrowable<UnsupportedFrameTypeException> {
+    override fun createCopy(): UnsupportedFrameTypeException? = UnsupportedFrameTypeException(frame).also {
+        it.initCause(this)
```

Comment:
Just to understand, do we actually need to call `it.initCause(this)`, not `it.initCause(this.cause)`?

Reply:
Yep, we should do it as the reflective approach in `kotlinx.coroutines`.